src/AutoRest.CSharp/Common/Input/InputTypes/InputDecoratedType.cs

```diff
@@ -0,0 +1,7 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace AutoRest.CSharp.Common.Input;
+internal record InputDecoratedType(IReadOnlyList<InputDecoratorInfo> Decorators);
```

Comment:
there is no point of adding this type here. We could just put this property in `InputType`.